lib/request_headers_middleware/delayed_job.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
```

Comment:
I'd really move this part into its own gem like I did for [logstasher-plugin-delayed](https://github.com/MarcGrimme/logstasher-plugin-delayed). Two reasons:
1. It's an unnecessary dep
2. It adds a dep to TaggedLogging
3. It adds a dep to X-Request-ID
that has no connection to request_headers_middleware